### ISSUE TITLE
Nothing happens when I rung ./loadibec ../openiboot.img3

### ISSUE THREAD

**Issue by zdavatz** [planetbeing/iphonelinux#8]
I put my iPhone into Recovery Mode on my Linux box (Turn off unconnected iPhone, hold Home button and connect USB to iPhone) and then run: ./loadibec ../openiboot.img3 but nothing happens. I have an 2G iPhone. The Recovery-Mode Screen does not go away. I just stays there.

**Comment by splatterb0y** [comment]
Keep holding home and lock button to hardreset it. It should boot normally then. Otherwise you have to restore it using the DFU-Mode.

**Comment by zdavatz** [comment]
Thanks! That is not the problem. I would like to install Android and I would expect the screen to change as I install openiboot. Or that is what the howto says.

**Comment by zdavatz** [comment]
The steps I am doing are: 
1. Put your phone into Recovery Mode (not DFU Mode).
2. Run sudo ./loadibec openiboot.img3
3. The openiboot menu should now be displayed on your phone. 

I get no openiboot menu.

According to: 

http://www.idroidproject.org/wiki/Installing_iDroid_%28Linux%29

**Comment by splatterb0y** [comment]
It works for me. I am using an iTouch 1G with fw 3.1.2 and blackra1n. 
We noticed some time ago that the jailbreak is important to load openiboot on the itouch 1g.

**Comment by zdavatz** [comment]
My phone is jailbroken with pwnage and Firmware version 3.1.2 (7D11). dmesg tells me: usb 1-3: USB disconnect, address 14
usb 1-3: new high speed USB device using ehci_hcd and address 15

**Comment by splatterb0y** [comment]
Maybe you want to try blackra1n jailbreak and test it again. I don't own an iPhone so sorry I can't really help you with that issue.

**Comment by zdavatz** [comment]
I could try installing openiboot from my Mac not my Linux. Can you point me to a Howto?

**Comment by splatterb0y** [comment]
Have you tried blackra1n?
You can compile loadibec and oibc for mac but you need to get a openiboot.img3 from linux, because it doesn't compile under mac (yet) .

**Comment by zdavatz** [comment]
How would I try blackra1n, my phone is already jailbroken?

**Comment by Anon0** [comment]
restore from itunes. Make sure that you download the same version that you already have installed( http://www.felixbruns.de/iPod/firmware/ ), put your phone in recovery mode and do a shift-click on the itunes restore button and select the fresh firmware package. After its been restored run blackra1n and then you're ready togo.

**Comment by zdavatz** [comment]
Ok, check this out in the latest kernel release: http://kerneltrap.org/mailarchive/linux-driver-devel/2010/3/31/6884240

There is a new iPhone driver in the Kernel that does not require a jailbroken iPhone! ;) Maybe that will help me. ;)

**Comment by zdavatz** [comment]
@Anon0 ok, I done as you advised and I installed 3.1.2 from Restore, then I ran Blackra1n. But now when I do sudo client/loadibec openiboot.img3 I still get the same result: No reaction. Nothing happens when I run that from my Linux box. Any hints?

**Comment by Anon0** [comment]
Im not sure if it will help ... but add a line to your makefile( at the top )-> PLATFORM=2G Then run 'make clean' and run make to rebuild.
I don't know if there is any specific 2G Code. I would need to read more code to find out. When it just doesn't load like that its most likely a config error.

Try running loadibec from the root openiboot dir .... 
sudo ./client/loadibec openiboot.img3

**Comment by zdavatz** [comment]
> Im not sure if it will help ... but add a line to your makefile( at the top )-> PLATFORM=2G Then run 'make clean' and run make to rebuild.

Sorry: What makefiles, what folder?

> sudo ./client/loadibec openiboot.img3

That is exactly what I am doing!

**Comment by zdavatz** [comment]
Ok, I did the following: makes no difference:

~/.software/iphonelinux/openiboot/client> vi Makefile 

~/.software/iphonelinux/openiboot/client> make clean

rm *.o
rm oibc
rm loadibec

~/.software/iphonelinux/openiboot/client> make

cc -DHAVE_GETEUID -I/opt/local-universal-10.4/include -c oibc.c -o oibc.o
cc -DHAVE_GETEUID -I/opt/local-universal-10.4/include oibc.o -L/opt/local-universal-10.4/lib -lusb -lpthread -lreadline -o oibc
cc -DHAVE_GETEUID -I/opt/local-universal-10.4/include -c loadibec.c -o loadibec.o
cc -DHAVE_GETEUID -I/opt/local-universal-10.4/include loadibec.o -L/opt/local-universal-10.4/lib -lusb-1.0 -o loadibec
cc -DHAVE_GETEUID -I/opt/local-universal-10.4/include -c linux.c -o linux.o
cc -DHAVE_GETEUID -I/opt/local-universal-10.4/include linux.o -lusb -lpthread -lncurses
 -o linux

~/.software/iphonelinux/openiboot/client> cd ..

~/.software/iphonelinux/openiboot> sudo ./client/loadibec openiboot.img3

Would love to run Android on my iPhone. I am on Linux not on VirtualBox (Linux) on Mac.

**Comment by zdavatz** [comment]
Ok, got openiboot installed from my Mac. But it is a bit sad that it does not work from Linux.

**Comment by zdavatz** [comment]
I followed this howto: http://androidalot.com/modding/how-to-install-android-on-an-iphone-2g/

**Comment by zdavatz** [comment]
I do not know if this is the right place, but after installing openiboot and then restarting the phone, my iPhone stops booting after the message: WELCOME TO OPENIBOOT

**Comment by Anon0** [comment]
Thats correct. You will now need a compiled linux for your phone. You can communicate with the phone via oibc.

**Comment by Anon0** [comment]
btw, when issues are resolved .. its good to close them too.

**Comment by Anon0** [comment]
I've been giving this issue some thought and I have a suspicion that your vm software( assuming a VM here) for your Linux installation probably wont allow you to get into the low level USB stuff or it may be improperly configured.

**Comment by zdavatz** [comment]
Thank you Anon0! My iPhone now boots Android, though without Multitouch driver. I now am using http://www.idroidproject.org/wiki/Dripwn to get the Driver into Android.img.gz. that tool also helped me to extract the driver from the phone.

**Comment by zdavatz** [comment]
Ok, Touchscreen is working. But I got no WIFI. Any hints?

**Comment by boremovic** [comment]
My iTouch 1G also doesn't respond to sudo ./loadibec, I've tried restoring my iTouch ( 3.1.2 ), it's jailbroken with redsn0w and all. I've installed everything from libusb to every other package needed. I don't understand what's the problem, 2 days ago I successfully installed openiboot on it, and it ran like a charm. In the mean time, i uninstalled it, using uninstall command in the openiboot console. Restored it and now it doesn't want to install again. Has anyone encountered anything similar?

**Comment by terinjokes** [comment]
Compiling and Installing onto an ipt1G works fine for me from the git repo. Try restoring from iTunes, rejailbreak and try flashing openiboot again.

**Comment by boremovic** [comment]
Never mind, I solved the problem by restoring and using blackra1n on my iPod Touch 1G, instead of redsn0w. Seems that fixed the issue and I was able to install openiboot normally.

**Comment by ghost** [comment]
I have the exact same issue but no access to a mac. Can anyone help me?
